src/main.nr

```diff
@@ -1,24 +1,155 @@
+    };
+    use dep::aztec::selector::compute_selector;
+
+    use dep::aztec::types::type_serialization::field_serialization::{ FieldSerializationMethods, FIELD_SERIALIZED_LEN};
+    use dep::aztec::types::type_serialization::aztec_address_serialization::{AztecAddressSerializationMethods, AZTEC_ADDRESS_SERIALIZED_LEN};
 
+    use dep::aztec::types::address::AztecAddress;
+
+
+    global BET_AMOUNT_SLOT = 1;
+    global DIVINITY_ADDRESS_SLOT = 2;
+    global PRIVATE_ORACLE_ADDRESS_SLOT = 3;
+    global HOUSE_ADDRESS_SLOT = 4;
+    global BETS_SLOT = 5;
+
+    // TODO: add slow update tree to make the switch from private to public
     struct Storage {
+        bet_amount: ImmutableSingleton<BetAmountNote, BET_AMOUNT_NOTE_LEN>,
+        divinity: ImmutableSingleton<AddressNote, ADDRESS_NOTE_LEN>,
+        private_oracle: ImmutableSingleton<AddressNote, ADDRESS_NOTE_LEN>,
+        house: ImmutableSingleton<AddressNote, ADDRESS_NOTE_LEN>,
+        bets: Set<BetNote, BET_NOTE_LEN>,
     }
 
     impl Storage {
         fn init(context: Context) -> Self {
             Storage {
+                bet_amount: ImmutableSingleton::new(
+                    context,
+                    BET_AMOUNT_SLOT,
+                    BetAmountNoteMethods,
+                ),
+                divinity: ImmutableSingleton::new(
+                    context,
+                    DIVINITY_ADDRESS_SLOT,
+                    AddressNoteMethods,
+                ),
+                private_oracle: ImmutableSingleton::new(
+                    context,
+                    PRIVATE_ORACLE_ADDRESS_SLOT,
+                    AddressNoteMethods,
+                ),
+                house: ImmutableSingleton::new(
+                    context,
+                    HOUSE_ADDRESS_SLOT,
+                    AddressNoteMethods,
+                ),
+                bets: Set::new(context, BETS_SLOT, BetNoteMethods)
             }
         }
     }
- 
+
     // Constructs the contract.
     #[aztec(private)]
-    fn constructor() {}
+    fn constructor(bet_amount: Field, divinity: AztecAddress, private_oracle: AztecAddress, house: AztecAddress) {
+        let mut bet_amount_note = BetAmountNote::new(bet_amount);
+        storage.bet_amount.initialize(&mut bet_amount_note, Option::none(), false);
+
+        let mut divinity_note = AddressNote::new(divinity);
+        storage.divinity.initialize(&mut divinity_note, Option::none(), false);
+
+        let mut private_oracle_note = AddressNote::new(private_oracle);
+        storage.private_oracle.initialize(&mut private_oracle_note, Option::none(), false);
+
+        let mut house_note = AddressNote::new(house);
+        storage.house.initialize(&mut house_note, Option::none(), false);
+    }
+
+    #[aztec(private)]
+    fn create_bet(bet: bool) {
+        let sender = AztecAddress::new(context.msg_sender());
+        let house = storage.house.get_note().address;
+
+        let mut user_bet: BetNote = BetNote::new(sender, bet);
```

Comment:
We should create only one bet (for the user) and then broadcast it to the house too. This will mean that both have the same nullifier and both can access it.

Reply:
This will help us because if either the house or the user can close the bet and nullify the whole bet. If we do two notes that only one party knows, there is no way for the other party to actually know that the bet was closed